lib/httparty.rb

```diff
@@ -599,14 +599,18 @@ def process_headers(options)
           options[:headers] = headers.merge(options[:headers])
         end
 
-        options[:headers] = Utils.stringify_keys(process_dynamic_headers(options[:headers]))
+        options[:headers] = Utils.stringify_keys(process_dynamic_headers(options))
       end
     end
 
-    def process_dynamic_headers(headers)
-      headers.each_with_object({}) do |header, processed_headers|
+    def process_dynamic_headers(options)
```

Comment:
Maybe it would be better to move it to a separate class? Something like `HeaderProcessor`. I don't really like that we do everything in a base class. Additionally, the method became more complex, so it's justified

Reply:
@TheSmartnik please preview code with extracted HeadersProcessor class